.changeset/modern-horses-poke.md

```diff
@@ -0,0 +1,24 @@
+---
+'@backstage/plugin-catalog-backend-module-msgraph': minor
+---
+
+Improved performance by reducing the number of calls being made to the Microsoft Graph API.
+
+- Increased the number of rows returned at a time from 100 to 999.
+- Load user data in batches when getting users by group membership, rather than making individual calls per user.
+
+Improved logging to provide more information as the task progresses.
+
+Added support for loading only users and only groups
+If you don't add any configuration for groups or users, then they won't be ingested.
```

Comment:
The package is currently at version 0.x.  If I make this patch a breakingchange, will that bump the  pakage to 1.x.?  IF so, I'm not sure this change is the one to bump it to 1.0.

The docs have always told people to add the configuration, so whilst this may technically be a breaking change, I expect it to have little practical breaking effect.